disperser/common/chunkstore/chunk_writer.go

```diff
@@ -0,0 +1,103 @@
+package chunkstore
+
+import (
+	"context"
+	"fmt"
+	"github.com/Layr-Labs/eigenda/common/aws/s3"
+	"github.com/Layr-Labs/eigenda/disperser"
+	"github.com/Layr-Labs/eigenda/encoding"
+	"github.com/Layr-Labs/eigenda/encoding/rs"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/consensys/gnark-crypto/ecc/bn254"
+)
+
+// ChunkCoefficientMetadata contains metadata about how chunk coefficients are stored.
+// Required for reading chunk coefficients using ChunkReader.GetChunkCoefficients().
+type ChunkCoefficientMetadata struct {
+	// The total size of file containing all chunk coefficients for the blob.
+	DataSize int
+	// The maximum fragment size used to store the chunk coefficients.
+	FragmentSize int
+}
+
+// ChunkWriter writes chunks that can be read by ChunkReader.
+type ChunkWriter interface {
+	// PutChunkProofs writes a slice of proofs to the chunk store.
+	PutChunkProofs(ctx context.Context, blobKey disperser.BlobKey, proofs []*encoding.Proof) error
+	// PutChunkCoefficients writes a slice of frames to the chunk store.
+	PutChunkCoefficients(
+		ctx context.Context,
+		blobKey disperser.BlobKey,
+		frames []*rs.Frame) (*ChunkCoefficientMetadata, error)
+}
+
+var _ ChunkWriter = (*chunkWriter)(nil)
+
+type chunkWriter struct {
+	logger       logging.Logger
+	s3Client     s3.Client
+	bucketName   string
+	fragmentSize int
+}
+
+// NewChunkWriter creates a new ChunkWriter.
+func NewChunkWriter(
+	logger logging.Logger,
+	s3Client s3.Client,
+	bucketName string,
+	fragmentSize int) ChunkWriter {
+
+	return &chunkWriter{
+		logger:       logger,
+		s3Client:     s3Client,
+		bucketName:   bucketName,
+		fragmentSize: fragmentSize,
+	}
+}
+
+func (c *chunkWriter) PutChunkProofs(ctx context.Context, blobKey disperser.BlobKey, proofs []*encoding.Proof) error {
+	s3Key := blobKey.String()
+
+	bytes := make([]byte, 0, bn254.SizeOfG1AffineCompressed*len(proofs))
+	for _, proof := range proofs {
+		proofBytes := proof.Bytes()
+		bytes = append(bytes, proofBytes[:]...)
+	}
+
+	err := c.s3Client.UploadObject(ctx, c.bucketName, s3Key, bytes)
```

Comment:
What is our s3 object versioning policy ? would we need to care for objects that already exist before uploading ?

Reply:
Discussed offline, repeating conclusion of that talk here for others.

Since each key is unique and has a deterministic value, writing a value to a key more than once is harmless (i.e. the data is overwritten with the exact same data).